### DISCUSSION TITLE
Resolving headless service pod across clusters without using clusterid

### DISCUSSION THREAD

**Discussion by abguptms** [submariner-io/lighthouse#1320] [General]
With the introduction of [Statefulset start ordinal](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#start-ordinal) in Kubernetes 1.27 onwards, there is a way to create a stateful application spanning multiple clusters with unique pod-ids in each cluster. However, to resolve the pod IPs across multiple clusters, we need to always be aware of which cluster a specific pod is part of. 
Meaning for clusterid-1 and clusterid-2, hosting pods 0-2 and 3-5 respectively for a statefulset, 
from clusterid-1, we always need to do something like this to resolve a pod in clusterid-2.
**statefulworkload-3.clusterid-2.statefulapp.my-namespace.svc.clusterset.local** (Similarly the other way around).

Since, the pods are unique in this structure throughout the group of clusters, can a way be provided to resolve a pod directly using:
**statefulworkload-3.statefulapp.my-namespace.svc.clusterset.local** 

Assumptions made here:
1. No two connected clusters are running the same ordinal index and range for the stateful applications.
2. Clusters are ephemeral, which means, stateful slices can move between clusters.
3. All clusters are connected via Submariner discovery (connectivity is inbuilt, gateway not required).
4. Only Statefulsets are using headless services.

Would using a different DNS:
**statefulworkload-3.statefulapp.my-namespace.pod.clusterset.local** help?

I wanted to get some thoughts on this. It's not necessarily an issue yet.

**Comment by abguptms** [comment]
@tpantelis, any thoughts on this?

**Reply by tpantelis** [reply]
Hmm...not off the top of my head - I'm not that familiar with StatefulSets.  cc @aswinsuryan @vthapar
